numba/typing/cffi_utils.py

```diff
@@ -24,13 +26,22 @@ def is_cffi_func(obj):
     try:
         return ffi.typeof(obj).kind == 'function'
     except TypeError:
-        return False
+        try:
+            return obj in _ool_func_types
+        except:
+            return False
 
 def get_pointer(cffi_func):
     """
     Get a pointer to the underlying function for a CFFI function as an
     integer.
     """
+    if cffi_func in _ool_func_types:
+        # We have no record of the function's address in an OOL module, so we
+        # resort to looking up the function by name in the interpreter's
+        # symbol table using ctypes
+        cfunc = getattr(ctypes.pythonapi, cffi_func.__name__)
```

Comment:
This only worked the test is using `sin` and `cos` symbols, which are in the "C" namespace.  This will not work for other user defined symbols; for example:

```python
from cffi import FFI

ffi = FFI()


ffi.cdef("int foo(int a, int b, int c);")

ffi.set_source("_example",
"""
static int foo(int a, int b, int c)
{
    return a + b * c;
}
""")


ffi.compile()

import _example

from numba import cffi_support, njit


cffi_support.register_module(_example)


foo = _example.lib.foo

print(foo(1, 2, 3))

@njit
def bar(x, y, z):
    return foo(x, y, z)

print(bar(1, 2, 3))

```